backend/lib/til/activities/reaction.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Til.Activities.Reaction do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Til.Accounts.User
+  alias Til.ShareableContent.Post
+
+  schema "reactions" do
+    field :type, :string
+
+    belongs_to :user, User
+    belongs_to :post, Post
+
+    timestamps()
+  end
+
+  def changeset(reaction, attrs) do
+    reaction
+    |> cast(attrs, [:user_id, :post_id, :type])
+    |> validate_required([:user_id, :post_id, :type])
+    |> validate_inclusion(:type, ["like", "funny", "love", "surprised"])
+    |> foreign_key_constraint(:post_id)
```

Comment:
should we check for user_id as well?

Reply:
Yes you are right, i used only post here cause i occured only problem with unexisting posts. It is better to secure it for future.